pantheon-sessions.php

```diff
@@ -359,14 +359,15 @@ public function add_index( $start_position ) {
 
 		$this->safe_output( __( 'Multisite installation detected. Processing Sites individually...', 'wp-native-php-sessions' ), 'log' );
 
-		$site_list = get_sites( [ 'number' => 500 ] );
+		$site_list = get_sites( [ 'number' => 99999 ] );
```

Comment:
I would advise against an arbitrary high number. I think you could keep the `500` and use `offset` for subsequent passes.

I also think you can optimize this query a bit by just returning `ids`, e.g. 

```php
$paged = 0; // Assume this gets incremented on each pass.
$num_sites = 500;
$site_list = get_sites( [
    'number' => $num_sites,
    'offset' => $paged * $num_sites,
    'fields' => 'ids',
] );
```

I don't think you need anything else in the WP_Site object besides the site ID (correct me if I'm wrong) but even if you did, you could specify those fields in `fields` and hit the database slightly less hard.

https://developer.wordpress.org/reference/functions/get_sites/ (click the More Arguments from WP_Site_Query toggle for all the parameters that can be passed in.)

It might actually be possible to get _all_ sites by passing `-1` to `number` as well, but that's also somewhat inadvisable for similar reasons. Better to batch process the sites rather than query thousands of sites in one go.

Another thought/option is that you could store the `$site_list` in cache or a transient to keep your place or if you needed that information again later. I don't think that would be super beneficial for `$site_list` but it might be useful for holding your place in the batch, e.g.:

```php
$offset = get_transient( 'wp_sessions_multisite_index_placeholder' );
if ( ! $offset ) {
    $offset = $paged * $num_sites;
    set_transient( 'wp_sessions_multisite_index_placeholder', $offset  );
}
```

That way, if the operation is stopped in the middle, you know in what batch you left off. 

Transient is just an idea but I think optimizing the `get_sites` query is going to be more important.